include/swift/AST/DiagnosticsSema.def

```diff
@@ -4121,6 +4121,9 @@ ERROR(asynchandler_async,none,
 ERROR(asynchandler_inout_parameter,none,
       "'inout' parameter is not allowed in '@asyncHandler' function",
       ())
+ERROR(asynchandler_unsafe_pointer_parameter,none,
+      "'%0' parameter is not allowed in `@asyncHandler` function",
```

Comment:
```suggestion
      "'%0' parameter is not allowed in '@asyncHandler' function",
```

Reply:
Ah, you're right. I considered banning it outright, then forgot to remove the diagnostic. Thank you!